addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/Base64UrlEncoderUnitTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.encoder.processors.predefined;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Test;
+import org.zaproxy.addon.encoder.processors.EncodeDecodeResult;
+
+public class Base64UrlEncoderUnitTest extends ProcessorTests<Base64UrlEncoder> {
+
+    @Override
+    protected Base64UrlEncoder createProcessor() {
+        return Base64UrlEncoder.getSingleton();
+    }
+
+    @Test
+    void shouldEncodeSimpleString() throws Exception {
+        // Given / When
+        EncodeDecodeResult result = processor.process("admin");
+        // Then
+        assertThat(result.hasError(), is(equalTo(false)));
+        assertThat(result.getResult(), is(equalTo("YWRtaW4=")));
+    }
+
+    @Test
+    void shouldEncodeStringWhichDoesntMatchStandardBase64() throws Exception {
```

Comment:
This could be tweaked as well.

Reply:
I'll change it to `shouldEncodeBase64UrlString`, the point I was trying to hit is that in this case the encoded value isn't the same as just base64 encoding (vs the `shouldEncodeSimpleString` case in which it is the same [base64 vs. base64Url).

Reply:
I see, I'd suggest tweak to e.g. `shouldEncode*As*Base64UrlString`, otherwise it seems it's testing the encode of a string that's already base64 URL encoded.